File: Codeforces rounds/Codeforces rounds/Codeforces Round 113 (Div. 2)/A.cpp
// https://codeforces.com/contest/166/problem/A

#include <bits/stdc++.h>

using namespace std;

typedef unsigned int uint;

bool cmp(pair<uint, uint> a, pair<uint, uint> b) {
  auto [pa, ta] = a;
  auto [pb, tb] = b;

  return pa > pb || (pa == pb && ta < tb);
}

uint solve(vector<pair<uint, uint>> pts, uint k) {
  sort(pts.begin(), pts.end(), cmp);

  auto [pk, tk] = pts[k - 1];

  uint ans = 0;

  for (auto [p, t] : pts) {
    if (p == pk && t == tk) {
      ans++;
    }
  }

  return ans;
}

int main(void) {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);

  uint n, k;
  cin >> n >> k;

  vector<pair<uint, uint>> pts(n);
  for (auto& [p, t] : pts) {
    cin >> p >> t;
  }

  auto ans = solve(pts, k);
  cout << ans << '\n';

  return EXIT_SUCCESS;
}

File: Codeforces rounds/Codeforces rounds/Codeforces Round 113 (Div. 2)/A.py
# https://codeforces.com/contest/166/problem/A

from functools import cmp_to_key

def cmp(a: tuple[int, int], b: tuple[int, int]) -> int:
    pa, ta = a
    pb, tb = b

    if pa > pb or (pa == pb and ta < tb):
        return -1
    elif pa == pb and ta == tb:
        return 0
    else:
        return 1


def solve(pts: list[tuple[int, int]], k: int) -> int:
    pts = sorted(pts, key=cmp_to_key(cmp))

    pk, tk = pts[k - 1]

    ans: int = 0

    for p, t in pts:
        if p == pk and t == tk:
            ans += 1

    return ans


def main():

    n, k = map(int, input().split())
    pts: list[tuple[int, int]] = []
    for _ in range(n):
        pts.append(tuple(map(int, input().split())))

    ans: int = solve(pts, k)
    print(ans)


if __name__ == '__main__':
    main()
